client/home/lib/account/deleteaccount/deleteaccountmodal.coffee

```diff
@@ -0,0 +1,60 @@
+kd = require 'kd'
+ContentModal = require 'app/components/contentModal'
+SingleGroupInfo = require './singlegroupinfo'
+whoami = require 'app/util/whoami'
+fetchMyRelativeGroups = require 'app/util/fetchMyRelativeGroups'
+VerifyPasswordModal = require 'app/commonviews/verifypasswordmodal'
+verifyPassword = require 'app/util/verifyPassword'
+showError = require 'app/util/showError'
+TeamFlux = require 'app/flux/teams'
+require('./styl/deleteaccount.styl')
+
+module.exports = class DeleteAccountModal extends ContentModal
+
+  constructor: (options, groups) ->
+
+    options = _.assign {}, options,
+      title : 'Account Managnment!'
+      overlay : yes
+      width : 700
+      cssClass : 'content-modal delete-account-modal'
+
+    super options
+
+    @main.addSubView new kd.CustomHTMLView
+      tagName: 'p'
+      partial: '
+        <p>
+          <strong>CAUTION!</strong>
+          You can not delete your account if you are owner of any team. You must
+          tranfer the ownership or delete the team.
+          These are the teams that you are owner of.
+        </p>
+        <p>
+          Transfering ownership makes you an admin of that team.
+          You must login and save your keys before you delete your account.
+        </p>
+        <p>
+          Delete team must be handled in the team so you will be redirected.
+        </p>'
+
+    @main.addSubView teamsWrapper = new kd.CustomHTMLView
+      cssClass: 'teamswrapper'
+
+    groups.forEach (group) ->
+      teamsWrapper.addSubView new SingleGroupInfo {}, group
+
+    @addSubView footer = new kd.CustomHTMLView
+      cssClass: 'modal-footer'
+
+    footer.addSubView deleteAccount = new kd.ButtonView
+      title          : 'Delete Team And Account'
+      cssClass       : 'GenericButton delete-account-button'
+      callback       : ->
+        fetchMyRelativeGroups (err, groups) ->
+
+          groups = groups.filter (group) -> 'owner' in group.roles
```

Comment:
we can also pass this query to the backend so we won't need to fetch all the groups that I'm member of which is not required for this case. number of groups that you are member of will always be more number of groups that you own, all of the time. so better to not fetch all that data over the wire.